test/cpp/end2end/xds/xds_core_end2end_test.cc

```diff
@@ -1247,6 +1247,8 @@ TEST_P(XdsMetricsTest, MetricValues) {
   const std::string kTarget = absl::StrCat("xds:", kServerName);
   const std::string kXdsServer = absl::StrCat("localhost:", balancer_->port());
   CreateAndStartBackends(1, /*xds_enabled=*/true);
+  ASSERT_TRUE(backends_[0]->notifier()->WaitOnServingStatusChange(
```

Comment:
I don't see how this helps, since the RPC is using wait_for_ready.

Reply:
Oh, wait -- I see from the test log that the problem is that the channel was in TRANSIENT_FAILURE, presumably because it took too long for the servers to come up.

Do we know why this is happening in these specific tests?  It seems like it could be a problem for *any* of the e2e tests.

Reply:
These two cases are where we are using xds enabled servers which can take slightly long to come up, but you are right, RPC using wait_for_ready would also work.

Reply:
done